src/freenet/support/MultiValueTable.java

```diff
@@ -1,230 +1,260 @@
-		}
+        return getAllAsList(key);
     }
 
     public int countAll(K key) {
-    	Vector<V> v;
-		synchronized (table) {
-			v = table.get(key);
-		}
-    	if(v != null)
-        	return v.size();
-        else
-        	return 0;
+        return getAllAsList(key).size();
     }
 
+    /**
+     * Returns mapped value collection as raw {@link Object}
+     *
+     * <b>Note:</b> this method is deprecated, this was used in previous {@code synchronized} implementation variant,
+     * please use other {@link #getAllAsList(Object)})} method variant, which provides a typed {@link List} collection.
+     *
+     * @param key key mapping
+     */
+    @Deprecated
     public Object getSync(K key) {
-		synchronized (table) {
-			return table.get(key);
-		}
+        return getAllAsList(key);
     }
 
+    /**
+     * Returns mapped value collection as array
+     *
+     * <b>Note:</b> this method is deprecated,
+     * please use other {@link #getAllAsList(Object)} method variant, which provides a typed {@link List} collection.
+     *
+     * @param key key mapping
+     */
+    @Deprecated
     public Object[] getArray(K key) {
-        synchronized (table) {
-            Vector<V> v = table.get(key);
-            if (v == null)
-                return null;
-            else {
-            	Object[] r = new Object[v.size()];
-                v.copyInto(r);
-                return r;
-            }
-        }
+        return getAllAsList(key).toArray();
     }
 
     public void remove(K key) {
-		synchronized (table) {
-			table.remove(key);
-		}
+        this.table.remove(key);
     }
 
     public boolean isEmpty() {
-		synchronized (table) {
-			return table.isEmpty();
-		}
+        return this.table.isEmpty();
+    }
+
+    public int size() {
+        return this.table.size();
     }
 
     public void clear() {
-		synchronized (table) {
-			table.clear();
-		}
+        this.table.clear();
     }
 
     public boolean removeElement(K key, V value) {
-        synchronized (table) {
-            Vector<V> v = table.get(key);
-            if (v == null)
-                return false;
-            else {
-                boolean b = v.removeElement(value);
-                if (v.isEmpty())
-                    table.remove(key);
-                return b;
+        boolean[] removed = new boolean[1];
+        this.table.computeIfPresent(key, (k, list) -> {
+            removed[0] = list.remove(value);
+            if (list.isEmpty()) {
+                // null result removes element from Map
+                return null;
             }
-        }
+            return list;
+        });
+        return removed[0];
     }
 
+    /**
+     * Returns table keys as {@link Enumeration}
+     *
+     * <b>Note:</b> this method is deprecated,
+     * please use other {@link #keySet()} method variant, which provides a {@link Set} of keys.
+     *
+     * @param key key mapping
+     */
+    @Deprecated
     public Enumeration<K> keys() {
```

Comment:
This method does not have a `key` parameter even though one is specified in the Javadoc.

Reply:
Fixed